engine/src/main/java/org/terasology/engine/rendering/opengl/GLSLMaterial.java

```diff
@@ -228,7 +228,8 @@ public void activateFeature(ShaderProgramFeature feature) {
             activeFeaturesMask = ShaderProgramFeature.getBitset(activeFeatures);
             activeFeaturesChanged = true;
         } else {
-            logger.error("Attempt to activate unsupported feature {} for material {} using shader {}", feature, getUrn(), shader.getUrn());
+            logger.atError().addArgument(feature).addArgument(() -> getUrn()).addArgument(() -> shader.getUrn()).
+                    log("Attempt to activate unsupported feature {} for material {} using shader {}", feature, getUrn(), shader.getUrn());
```

Comment:
Only evaluated in unexpected cases. Evaluated methods are simple getters for member access so impact neglectable.
```suggestion
            logger.error("Attempt to activate unsupported feature {} for material {} using shader {}", feature, getUrn(), shader.getUrn()); //NOPMD
```